planner/core/find_best_task.go

```diff
@@ -1033,6 +1036,10 @@ func (ds *DataSource) getOriginalPhysicalTableScan(prop *property.PhysicalProper
 	if ds.preferStoreType&preferTiKV != 0 {
 		ts.StoreType = kv.TiKV
 	}
+	if infoschema.IsClusterMemTable(ds.DBName.L, ds.tableInfo.Name.L) {
+		ts.StoreType = kv.TiDBMem
+		ts.tp = plancodec.TypeMemTableScan
```

Comment:
This is pretty strange, can we keep using `TypeTableScan`?

Reply:
I think we should not define the `MemTableScan`. We can check whether a table is a virtual table in `buildPlan` from protobuf and build a `PhysicalMemTable` directly.

Reply:
Great, done.